### PR TITLE
feat(ci): Upload telemetry to Kusto in test-dds-stress pipeline

### PR DESCRIPTION
**Pull request by clarenceli-msft**
[AB#2731](https://dev.azure.com/fluidframework/internal/_workitems/edit/2731)

Move notification mechanisms of `Test - DDS Stress` pipeline to use telemetry alerting rather than ADO notifications.

The updates of YAML file was verified successful on the test branch: https://dev.azure.com/fluidframework/internal/_build/results?buildId=270864&view=results

Create a new Kusto-to-Metrics query: https://portal.microsoftgeneva.com/?page=settings&mode=connectors&tab=kusto&activity=kusto-to-metrics&version=1&account=FluidFrameworkMetrics&namespace=PipelineHealth&ruleId=f676f037-c722-4d15-b42a-024b8a23e9da

Set the monitor to track the consecutive failures of this pipeline: https://portal.microsoftgeneva.com/manage/monitors/monitor?activity=edit-monitor&version=1&action=1&account=FluidFrameworkMetrics&newManageSessionId=176b98d3-eaf7-4a97-bc0e-e976a76d3fb1

### CODE REVIEW

File: tools/pipelines/test-dds-stress.yml [hunk 1]
```diff
@@ -22,38 +22,87 @@ resources:
...
+      packages: ${{ parameters.packages }}
       testWorkspace: ${{ variables.testWorkspace }}
+
+  # Capture telemetry about pipeline stages
+  - stage: FollowupStage
+    displayName: Upload pipeline run telemetry to Kusto
+    condition: succeededOrFailed()
+    dependsOn:
+    - ${{ each package in parameters.packages }}:
+      - ${{ replace(replace(package.name, '@fluidframework/', ''), '@fluid-experimental/', 'experimental_') }}_stress_tests
```
Comment (alexvy86) [inline tools/pipelines/test-dds-stress.yml:76]:
```suggestion
      # This list of stages should be kept in sync with the way we generate their ids in tools/pipelines/templates/include-conditionally-run-stress-tests.yml
      - ${{ replace(replace(package.name, '@fluidframework/', ''), '@fluid-experimental/', 'experimental_') }}_stress_tests
```

File: tools/pipelines/templates/include-conditionally-run-stress-tests.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ stages:
             AFFECTED_PATHS: ${{ join(';', package.affectedPaths) }}
 
   - ${{ each package in parameters.packages }}:
-    - stage:
+    - stage: ${{ replace(replace(package.name, '@fluidframework/', ''), '@fluid-experimental/', 'experimental_') }}_stress_tests
```
Comment (alexvy86) [inline tools/pipelines/templates/include-conditionally-run-stress-tests.yml:63]:
```suggestion
    # The ids for these stages should be kept in sync with the dependency on them in tools/pipelines/test-dds-stress.yml
    - stage: ${{ replace(replace(package.name, '@fluidframework/', ''), '@fluid-experimental/', 'experimental_') }}_stress_tests
```

File: tools/pipelines/test-dds-stress.yml [hunk 3]
```diff
@@ -22,38 +22,87 @@ resources:
...
-        affectedPaths:
-        - packages/dds/matrix
-        - packages/dds/merge-tree
-        testFileTarName: matrix
-        testCommand: test:stress
+      packages: ${{ parameters.packages }}
       testWorkspace: ${{ variables.testWorkspace }}
+
+  # Capture telemetry about pipeline stages
+  - stage: FollowupStage
```
Comment (alexvy86) [inline tools/pipelines/test-dds-stress.yml:71]: Nit, for consistency with the other pipelines
```suggestion
  - stage: upload_run_telemetry
```

**Review by alexvy86 — APPROVED** [review]
A few nits below. The kusto-to-metrics query and monitor look good, but two things:

- Since the stages for each package are dynamic, I feel a bit uneasy hardcoding them in the query. In fact, I'm not convinced we should do any stage filtering in any of the queries. We can address the existing ones separately, but I'd say for this new one let's remove the line that filters based on stage.
- Before enabling the monitor, we need to have a new troubleshooting guide in our internal docs for this pipeline so OCE knows what to do if we get an incident for that monitor. See [the current one for the stress tests pipeline](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/on-call/ff-hot/pipelinemonitoring/stress-test-pipeline).

**Comment by alexvy86** [comment]
> Before enabling the monitor, we need to have a new troubleshooting guide in our internal docs for this pipeline so OCE knows what to do if we get an incident for that monitor.

Particularly important since it seems https://github.com/microsoft/FluidFramework/pull/20815 broke the tree stress tests and that stage of the pipeline is failing :). FYI @Abe27342 .

**Comment by clarenceli-msft** [comment]
> A few nits below. The kusto-to-metrics query and monitor look good, but two things:
> 
> * Since the stages for each package are dynamic, I feel a bit uneasy hardcoding them in the query. In fact, I'm not convinced we should do any stage filtering in any of the queries. We can address the existing ones separately, but I'd say for this new one let's remove the line that filters based on stage.
> * Before enabling the monitor, we need to have a new troubleshooting guide in our internal docs for this pipeline so OCE knows what to do if we get an incident for that monitor. See [the current one for the stress tests pipeline](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/on-call/ff-hot/pipelinemonitoring/stress-test-pipeline).

Good call, I will update the TSG, tracked in this work item [AB#7359](https://dev.azure.com/fluidframework/internal/_workitems/edit/7359)

**Pull request merged**